File: scripts/examples/seir_simulation.jl
using DifferentialEquations
using Plots

function seir_model!(du, u, p, t)
    S, E, I, R = u
    beta, sigma, gamma = p
    N = S + E + I + R

    du[1] = -beta * S * I / N
    du[2] = beta * S * I / N - sigma * E
    du[3] = sigma * E - gamma * I
    du[4] = gamma * I
end

u0 = [990.0, 10.0, 0.0, 0.0]
p = (0.5, 0.2, 0.1) # beta, sigma, gamma
tspan = (0.0, 160.0)
prob = ODEProblem(seir_model!, u0, tspan, p)
sol = solve(prob)
plot(sol)
savefig("assets/seir_plot.png")

File: scripts/examples/seir_simulation.py
import numpy as np
from scipy.integrate import odeint
import matplotlib.pyplot as plt


def seir_model(u, t, beta, sigma, gamma):
    S, E, I, R = u
    N = S + E + I + R

    dSdt = -beta * S * I / N
    dEdt = beta * S * I / N - sigma * E
    dIdt = sigma * E - gamma * I
    dRdt = gamma * I
    return [dSdt, dEdt, dIdt, dRdt]


u0 = [990, 10, 0, 0]
t = np.linspace(0, 160, 160)
beta, sigma, gamma = 0.5, 0.2, 0.1

sol = odeint(seir_model, u0, t, args=(beta, sigma, gamma))
plt.plot(t, sol)
plt.legend(["S", "E", "I", "R"])
plt.title("SEIR Model Simulation")
plt.xlabel("Time")
plt.ylabel("Population")
plt.savefig("assets/seir_plot_python.png")
